src/models/base_model.rs

```diff
@@ -181,6 +261,40 @@ impl BaseModel {
     }
 }
 
+pub struct PathwayByIter<'model> {
+    idx_iter: Option<std::collections::hash_set::Iter<'model, (PathwayIdx, StopLocationIdx)>>,
+    model: &'model BaseModel,
+}
+
+impl<'model> PathwayByIter<'model> {
+    pub fn new(stop_point_idx: &BaseStopPointIdx, model: &'model BaseModel) -> Self {
+        let idx_iter = model
+            .stop_point_to_pathways
+            .get(stop_point_idx)
+            .map(|pathways| pathways.iter());
+        Self { idx_iter, model }
+    }
+}
+
+impl<'model> Iterator for PathwayByIter<'model> {
+    type Item = (&'model Pathway, StopLocationIdx);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let iter = self.idx_iter.as_mut()?;
+        let (pathway_idx, access_point_idx) = iter.next()?;
+        Some((
+            self.model.model.pathways.index(*pathway_idx),
+            *access_point_idx,
+        ))
+    }
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match &self.idx_iter {
+            Some(iter) => iter.size_hint(),
+            None => (0, None),
```

Comment:
```suggestion
            None => (0, 0),
```
You know that the iterator is empty when idx_iter is None